### ISSUE TITLE
v3: Notification-API does not work fully?!

### ISSUE THREAD

**Issue by RkcCorian** [Zefau/ioBroker.jarvis#1095]
*Please note that you may restore an old layout via jarvis adapter configuration directly within ioBroker

**Prerequisites / Voraussetzungen**
- [x] Which jarvis version are you running / Welche Version von jarvis nutzt du? **BITTE GENAUE ANGABE!**
- [x] Fehler / Screenshot der Browser Konsole (F12), siehe [Fehleranalyse und -Meldung](https://github.com/Zefau/ioBroker.jarvis/wiki/de-Errors)

```alpha-86```

**Describe the bug / Beschreibe den Fehler**
A clear and concise description of what the bug / problem is.
Eine genaue Beschreibung des Fehlers inkl. Herleitung, um diesen Nachstellen zu können.

Ich starte nun mit Benachrichtigungen. Die Basics klappen soweit mit v3-alpha-86, allerdings nicht alles. Bevor ich Issues erstelle... wie ist der letzte Stand der API? Ich versuche folgendes Beispiel...
```
{"title":"TestTitle", "message":"17:16:37 Uhr | 09.10.21 | Dies ist der Text", "state":"unread", "criticality":"warnung", "icon":"home", "iconColor":"#ff0000", "fontColor":"#ff0000", "backgroundColor":"#000000", "display":"popup", "autoHideDuration":"5000", "focusTab":"f0c1c414-de3c-42b1-a5ba-5ac9dd91a5b9"}
```
Keinen Effekt haben
- criticality
- iconColor
- fontColor
- backgroundColor
- focusTab

Sollten die einen Effekt haben?!

**Screenshots**
Sofern möglich, bitte Screenshots einfügen, um den Fehler bildlich zu erläutern.

____

**Please note that bug reports may be closed if the information above are not given!**

**Bitte beachte, dass Bug-Reports ggf. einfach geschlossen werden, wenn die oben genannten Kriterien nicht angemessen gegeben sind!**

**Comment by mcuiobroker** [comment]
Stimmt, musste doch glatt "iobroker upload jarvis" durchführen. Dann teste ich die anderen Sachen auch nochmal.

**Comment by mcuiobroker** [comment]
`v3.0.0-alpha.90` bekomme jetzt die snackbar-Notification mehrfach angezeigt, obwohl nur einmal geschickt?
![image](https://user-images.githubusercontent.com/84818934/136852750-d9001d2d-2edd-4a8d-aeb6-e74e73139989.png)

**Comment by Zefau** [comment]
Mit `v3.0.0-alpha.119` gibt es nun die `delete` Option. Außerdem ist `category` nun als CSS Selektor in der Sidebar verfügbar.

**Comment by RkcCorian** [comment]
Bin bereit :-) Welche ```category``` gibt es jetzt?

**Comment by Zefau** [comment]
Die, die du in der Notification definierst. Die kannst du in der Sidebar zum stylen nutzen.

**Comment by mcuiobroker** [comment]
Option "delete" funktioniert nicht, sie bleiben als "unread" hängen.

**Comment by RkcCorian** [comment]
@Zefau: Wie kann ich die Icon-Farbe im NotificationCenter über CSS beeinflussen?
![image](https://user-images.githubusercontent.com/79741077/140663465-17ed8c41-e802-4c92-a2db-5bfe060a9022.png)

**Comment by RkcCorian** [comment]
Ich bin mir immer noch nicht sicher, was sich jetzt geändert hat, evtl. blicke ich es nicht 🙈. Folgendes CSS hatte ich auch vor alpha-119 bereits für die Category ```notificationAus``` benutzt.

```
/* Benachrichtigungen (Alle) ======================================== */

.jarvis-notification-event-popup-actions {
	/* Hintergrundfarbe Buttons */
	background-color: #c7c7c7 !important;
}

.jarvis-notification-event-popup-actions .q-btn {
	/* Schriftfarbe Buttons - Archiv - Mark as read */
	color: #242424;
}

.jarvis-notification-event-popup-title .jarvis-icon svg {
	color: #ffffff !important;
}

/* Benachrichtigungen (notificationAus) =============================== */

.jarvis-notification-event-popup.notificationAus {
	/* Schriftfarbe Text */
	color: #ffffff !important;
	/* Hintergrundfarbe Allgemein */
	background-color: #424242 !important;
}

.jarvis-notification-event-popup.notificationAus .jarvis-notification-event-popup-title {
	/* Hintergrundfarbe Titel */
	background-color: #424242 f!important;
}

.jarvis-notification-event-popup-title.notificationAus .jarvis-icon svg {
	color: #b366ff !important;
}

.jarvis-notification-event-popup.notificationAus .jarvis-notification-event-popup-message {
	/* Hintergrundfarbe Nachricht */
	background-color: #424242 !important;
}
```

**Comment by mcuiobroker** [comment]
Das war doch das Thema mit "test".
.jarvis-notification-event-popup.Test OK
.jarvis-notification-event-popup.Test .jarvis-notification-event-popup-title OK
.jarvis-notification-event-popup.Test .jarvis-notification-event-popup-message OK
.jarvis-notification-event-popup.Test .jarvis-notification-event-popup-actions NOK > Keine Auswirkung Hätte erwartet dass dies die Farbe der Button (Read, Archive) beeinflusst

**Comment by Zefau** [comment]
Angenommen deine Kategory heißt `test`, dann:

```
.jarvis-notifications-item.test {
   ...
}
```

für das Icon

```
.jarvis-notifications-item.test .iconify {
   color: #f000
}
```

**Comment by Zefau** [comment]
>  .jarvis-notification-event-popup.Test .jarvis-notification-event-popup-actions NOK > Keine Auswirkung Hätte erwartet dass dies die Farbe der Button (Read, Archive) beeinflusst

`actions` verweist auf das `div`, nicht die Buttons. Die Buttons musst du dann ansteuern. Einfach im Insektor schauen.

**Comment by mcuiobroker** [comment]
Ich hatte es nur von oben kopiert.

**Comment by RkcCorian** [comment]
Auf welches Icon sollte 
```
.jarvis-notifications-item.test .iconify {
   color: #f000
}
```
einen Einfluss haben? Benachrichtigung und/oder Benachrichtungungszentrale

**Comment by Zefau** [comment]
Benachrichtungungszentrale. Aber habe es nicht getestet. Bitte wegen den CSS Selektoren selbst schauen.

**Comment by RkcCorian** [comment]
> Das war doch das Thema mit "test". .jarvis-notification-event-popup.Test OK .jarvis-notification-event-popup.Test .jarvis-notification-event-popup-title OK .jarvis-notification-event-popup.Test .jarvis-notification-event-popup-message OK .jarvis-notification-event-popup.Test .jarvis-notification-event-popup-actions NOK > Keine Auswirkung Hätte erwartet dass dies die Farbe der Button (Read, Archive) beeinflusst

Ahh, ok. Das mache ich jetzt für alle Benachrichtigungen konsistent über 
```
.jarvis-notification-event-popup-actions .q-btn {
	/* Schriftfarbe Buttons - Archiv - Mark as read */
	color: #242424;
}
```

**Comment by RkcCorian** [comment]
> Benachrichtungungszentrale. Aber habe es nicht getestet. Bitte wegen den CSS Selektoren selbst schauen.

Wollte nur sicher gehen, da ich keinen Effekt sehe. D.h. es klappt nicht.

**Comment by RkcCorian** [comment]
Gibt es auch ein ```.jarvis-notification-event-popup-caption```? Oder sollte es eines geben?!

**Comment by RkcCorian** [comment]
> Benachrichtungungszentrale. Aber habe es nicht getestet. Bitte wegen den CSS Selektoren selbst schauen.

Hab jetzt ein paar Sachen ausprobiert, bekomme es aber leider nicht hin. Evtl. blicke ich es aber auch nicht.

**Comment by RkcCorian** [comment]
Update: Icons in der Benachrichtigungszentrale gehen! Was bei mir gefehlt hat was das ```!important``` 🙃 Also...

```
.jarvis-notifications-item.notificationAus .iconify {
	color: #ff0000 !important;
}
```
funktioniert

**Comment by RkcCorian** [comment]
Kann jetzt Icon-Farbe sowohl von der Benachrichtigungszentrale als auch der Benachrichtigung getrennt steuern. Super!!! Anbei mein Dummy-CSS.

```
/* Benachrichtigungen (Alle) ======================================== */

.jarvis-notification-event-popup-title .jarvis-icon svg {
	/* Icon-Farbe Benachrichtigung */
	/*color: #ffffff !important;*/
}

/* Benachrichtigungen (notificationAus) =============================== */

.jarvis-notifications-item.notificationAus .iconify {
	/* Icon-Farbe Benachrichtigungszentrale */
	color: #ff0000 !important;
}

.jarvis-notification-event-popup.notificationAus .jarvis-icon svg {
	/* Icon-Farbe Benachrichtigung */
	color: #ff9900 !important;
}
```

**Comment by RkcCorian** [comment]
Verbleibt nur noch evtl. ```.jarvis-notification-event-popup-caption```. Wird es da was geben?

**Comment by Zefau** [comment]
Der Untertitel ist im selben `div` wie im Titel.
Also `.jarvis-notification-event-popup-title .text-caption`. Der Titel selbst ist  `.jarvis-notification-event-popup-title .text-h6`.

**Comment by Zefau** [comment]
@mcuiobroker ich hab die Übersicht verloren: Mit `v3.0.0-alpha.120` sollte der `delete` Modus passen.

**Comment by mcuiobroker** [comment]
Ja ```delete``` passt, Danke.
